### PR TITLE
PLAT-32: Create a Login OWA for the platform

### PR DESCRIPTION
**Pull request by Charpell**
## JIRA TICKET NAME
[PLAT-32: Create a Login OWA for the platform ](https://issues.openmrs.org/browse/PLAT-32)

## SUMMARY:
Currently, the platform runs independent of any login functionality.
An adjustment was made to bundle AddOn Manager with the platform and have the platform use the AddOn Manager's login functionality to authenticate users

The objective is to create an OWA called Platform UI handling the user authentication functionality that will then be used as the primary/default login OWA for the OpenMRS platform

## Issue I worked on
[PLAT-32](https://issues.openmrs.org/browse/PLAT-32)

## Checklist: I completed these to help reviewers
 - [x] My pull request only contains ONE single commit (the number above, next to the 'Commits' tab is 1).
No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [code style](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.
No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

 - [ ] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)
No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and added all formatting changes to my commit.
No? -> execute above command

- [ ] All new and existing **tests passed**.
No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

 - [x] My pull request is **based on the latest changes** of the master branch.
No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+  "name": "openmrs-owa-platformui",
+  "version": "0.1.0",
+  "description": "A very simple UI for the OpenMRS Platform to give access to the Addon Manager, User manager, and other tools.",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/charpell/openmrs-owa-openmrs-owa-platformui"
```
Comment (dkayiwa) [inline package.json]: Can we change this url to that of openmrs?

File: app/manifest.webapp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "version": "0.1.0",
+  "name": "openmrs-owa-platformui",
+  "description": "A very simple UI for the OpenMRS Platform to give access to the Addon Manager, User manager, and other tools.",
+  "launch_path": "index.html",
+  "icons": {
+    "48": "/img/omrs-button.png"
+  },
+  "developer": {
+    "name": "charpell",
+    "url": "https://github.com/charpell/openmrs-owa-openmrs-owa-platformui"
```
Comment (dkayiwa) [inline app/manifest.webapp]: Can we change this url to that of openmrs?

File: app/js/components/presentational/LoginForm.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import openmrsLogo from '../../../img/gsoc-omrs.png';
+
+const LoginForm = ({ onSubmit, onChange, username, password, sessionLocation }) => {
+  return (
+    <div className="row">
+      <div className="col-lg-10">
+        <form className="login-form" onSubmit={onSubmit}>
+          <h4 className="login-header">Welcome</h4>
+          <div className="logo-container">
+            <img className="logo-img" src={(openmrsLogo)} alt="Add On manager homepage" />
```
Comment (dkayiwa) [inline app/js/components/presentational/LoginForm.jsx]: Can we use capital M for manager?

Comment (dkayiwa) [reply app/js/components/presentational/LoginForm.jsx]: Instead of homepage, use home page

File: package.json (hunk 1 above)
Comment (Charpell) [reply package.json]: I have updated the PR

File: app/manifest.webapp (hunk 2 above)
Comment (Charpell) [reply app/manifest.webapp]: I have updated the PR

File: app/js/components/presentational/LoginForm.jsx (hunk 3 above)
Comment (Charpell) [reply app/js/components/presentational/LoginForm.jsx]: I have updated the PR

File: app/js/components/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            <h5 className="mb-1">The Platform is running</h5><img src={(loadingGif)} alt=""/>
+            </div>
+            
+          </a>
+          <a href="#" className="list-group-item list-group-item-action flex-column align-items-start">
+            <div className="d-flex w-100 justify-content-between">
+              <h5 className="mb-1">Addon Manager</h5>
+              
+            </div>
+            <p className="mb-1">The Addon manager open web app, is an OpenMRS tool used to manage the uploading, installation, upgrading, deleting and viewing of OpenMRS addons.</p>
```
Comment (dkayiwa) [inline app/js/components/Home.jsx]: Same here for addon manager

File: app/js/components/presentational/LoginForm.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (justmesam) [inline app/js/components/presentational/LoginForm.jsx:9]: All OpenMRS app files should have the license comments on top.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{
+  "name": "openmrs-owa-platformui",
+  "version": "0.1.0",
+  "description": "A very simple UI for the OpenMRS Platform to give access to the Addon Manager, User manager, and other tools.",
```
Comment (dkayiwa) [inline package.json]: User manager should also be capital M for manager

**Review by justmesam — COMMENTED** [review]
check on the comment, else looks good 👍

**Comment by patlub** [comment]
Except for the comments, Looks okay.

File: app/js/components/presentational/LoginForm.jsx (hunk 5 above)
Comment (Charpell) [reply app/js/components/presentational/LoginForm.jsx:9]: Great the PR had been updated

File: app/js/components/Home.jsx (hunk 4 above)
Comment (Charpell) [reply app/js/components/Home.jsx]: PR had been updated

File: package.json (hunk 6 above)
Comment (Charpell) [reply package.json]: PR had been updated

File: app/manifest.webapp [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+  "version": "0.1.0",
+  "name": "openmrs-owa-platformui",
+  "description": "A very simple UI for the OpenMRS Platform to give access to the Addon Manager, User manager, and other tools.",
```
Comment (dkayiwa) [inline app/manifest.webapp]: Can we use capital M for User manager

Comment (Charpell) [reply app/manifest.webapp]: Yes, i have updated it

File: app/js/components/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
```
Comment (justmesam) [inline app/js/components/Login.jsx:1]: You can also add license comments on this

**Pull request merged**